torch/_dynamo/backends/distributed.py

```diff
@@ -329,27 +312,46 @@ def compile_fn(self, gm: fx.GraphModule, example_inputs: List[torch.Tensor]):
         debug_str += "\n---------------\n"
         ddp_graph_log.debug(debug_str)
 
-        # 3: compile each of the partitioned submodules using the user-provided compiler
+        # 3: create a wrapper which lazily compiles each of the partitioned submodules
+        # using the user-provided compiler
         class SubmodCompiler(torch.fx.interpreter.Interpreter):
             def __init__(self, module, compiler):
                 super().__init__(module)
                 self.compiler = compiler
 
-            def compile_submod(self, input_mod, args, kwargs):
+            def lazily_compile_submod(self, input_mod):
                 """
                 Compile the submodule,
                 using a wrapper to make sure its output is always a tuple,
                 which is required by AotAutograd based compilers
                 """
-                assert len(kwargs) == 0, "We assume only args for these modules"
 
                 class WrapperModule(torch.nn.Module):
-                    def __init__(self, submod, unwrap_singleton_tuple):
+                    def __init__(self, submod, compiler, unwrap_singleton_tuple):
                         super().__init__()
                         self.submod = submod
+                        self.compiler = compiler
+                        self.compiled = False
                         self.unwrap_singleton_tuple = unwrap_singleton_tuple
 
                     def forward(self, *args):
+                        fake_args = []
+                        for arg in args:
+                            if isinstance(arg, torch.Tensor) and not isinstance(
+                                arg, torch._subclasses.FakeTensor
```

Comment:
Maybe we also want to handle fakeified? 

not `torch._subclasses.fake_tensor.is_fake(..)`